public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -388,6 +388,16 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     }
   }
 
+  onOpenSnapshotOriginalDashboard = () => {
```

Comment:
I think we could reuse the function from DashNav
https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/DashNav/DashNav.tsx#L88, there are some security checks we need :)

Reply:
Yes, I saw them and tried to reproduce the opening of the modal and I couldn't. There is a condition that never will be true `sanitizedUrl.host !== appUrl.host` because the `sanitizedUrl` is made with the same host `config.appUrl`. I think this is an old code that does not work anymore but I will need to dig more into this.

Reply:
Good catch! I've added these checks to the new code.